frontend/src/components/auth/LoginCard.vue

```diff
@@ -0,0 +1,138 @@
+<template>
+  <div class="login-box">
+    <div class="header-box">
+      <p class="header-text" id="login-card-title">Welcome Back to FLYSPRAY</p>
+      <p class="sub-header-text" id="login-card-title">
+        Web-based bug tracker system
+      </p>
+    </div>
+    <div class="form-box">
+      <p class="signin-text">Sign in to start managing your projects</p>
+      <input-form>
+        <v-sheet width="300" class="mx-auto">
+          <v-form ref="form" @submit.prevent="submitForm">
+            <v-text-field
+              prepend-inner-icon="mdi-email"
+              label="Email"
+              v-model="email"
+              required
+              :rules="[validateEmail]"
+            ></v-text-field>
+            <v-text-field
+              type="password"
+              prepend-inner-icon="mdi-lock"
+              label="Password"
+              v-model="password"
+              required
+              :rules="[validatePassword]"
+            ></v-text-field>
+
+            <div class="forgot-password-container">
+              <router-link to="/forget" class="link"
+                >Forgot Password?</router-link
+              >
+            </div>
+            <v-btn
+              type="submit"
+              block
+              class="mt-2 btn"
+              :disabled="errorPassword || errorEmail"
+              >Sign In</v-btn
+            >
+          </v-form>
+          <hr class="form-separator" />
+
+          <p class="signin-text">
+            Don't have account? please
+            <router-link to="/signup" class="link">Sign Up</router-link>
+          </p>
+        </v-sheet></input-form
+      >
+    </div>
+  </div>
+</template>
+
+<script lang="ts">
+import { defineComponent } from "vue";
+import InputForm from "./InputForm.vue";
+
+export default defineComponent({
+  components: {
+    InputForm,
+  },
+  data() {
+    return {
+      email: "" as string,
+      password: "" as string,
+    };
+  },
+  computed: {
+    errorEmail(): boolean {
+      const pattern = /^[a-zA-Z0-9._%+-]+@[a-zA-Z0-9.-]+\.[a-zA-Z]{2,}$/;
+      return !pattern.test(this.email);
+    },
+    errorPassword(): boolean {
+      return this.password.length < 8;
+    },
+  },
+  methods: {
+    validateEmail(value: string) {
+      if (!value) return "Email is required";
+      const pattern = /^[a-zA-Z0-9._%+-]+@[a-zA-Z0-9.-]+\.[a-zA-Z]{2,}$/;
+      return pattern.test(value) || "Invalid email format";
+    },
+    validatePassword(value: string) {
+      if (!value) return "Password is required";
+      return value.length >= 8 || "Password must be at least 8 characters long";
+    },
```

Comment:
in the future we'll have huge validations, What do think if we moved these functions to a separate file called `validations.ts` inside the `utils` folder

also the the validation should return two values (bool -> if valid , errorMessage if there is an error)

e.g.

```ts

type ValidationResult = {
  isValid: boolean;
  errorMessage?: string;
};

export function validateEmail(value: string): ValidationResult {
  if (!value || value.trim() === "") {
    return {
      isValid: false,
      errorMessage: "Email is required",
    };
  }

  // Implement email validation logic here
  // Example: Check if the value follows a valid email format
  // You can use regular expressions or any other validation logic

  if (!isValidEmailFormat(value)) {
    return {
      isValid: false,
      errorMessage: "Invalid email format",
    };
  }

  // Add more validation rules as needed

  return {
    isValid: true,
  };
}
```